### PR TITLE
Refactor lib/private/App

### PR DESCRIPTION
**Pull request by shdehnavi**
## Summary
The required adjustments have been made to the following classes in the `/lib/private/app ` namespace.

The improvements:

- Using PHP8's constructor property promotion

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits

### CODE REVIEW

**Comment by shdehnavi** [comment]
Hey @szaimen 
The following PRs haven't been assigned to any reviewers yet.
Could you please be so kind and assign them to reviewers? bc I want to make the new PRs and still I have 30 open PRs.
#39249 
#39246
#39245
#39244
#39243
#39236
#39235
#39226
#39217
#39216
#39215
#39213
#39212
#39211
#39209
#39208
#39183
#39182
#39177
#39176
#39175
#39174
#39173
#39163

**Comment by szaimen** [comment]
Hi @shdehnavi thanks a lot for the many PRs! Can you chwck if you can request the teview itself? Usually you should be able to reauest reviews from the nextcloud/server-backend group. Does that work for you?

**Comment by shdehnavi** [comment]
> Hi @shdehnavi thanks a lot for the many PRs! Can you check if you can request the review itself? Usually you should be able to request reviews from the nextcloud/server-backend group. Does that work for you?

sure, i will do it

File: lib/private/App/AppStore/Fetcher/AppFetcher.php [hunk 1]
```diff
@@ -39,22 +39,21 @@
...
-								LoggerInterface $logger,
-								IRegistry $registry) {
+	public function __construct(
+		Factory $appDataFactory,
+		IClientService $clientService,
+		ITimeFactory $timeFactory,
+		IConfig $config,
+		private CompareVersion $compareVersion,
+		LoggerInterface $logger,
+		IRegistry $registry,
```
Comment (come-nc) [inline lib/private/App/AppStore/Fetcher/AppFetcher.php:55]:
```suggestion
		protected IRegistry $registry,
```
Why is this one not handled the same way?

Comment (shdehnavi) [reply lib/private/App/AppStore/Fetcher/AppFetcher.php:55]: It's addressed in the next commit

**Review by come-nc — APPROVED** [review]

**Comment by fsamapoor** [comment]
Could you please resolve the conflicts?

File: lib/private/App/AppManager.php [hunk 2]
```diff
@@ -103,20 +95,16 @@
...
-		$this->memCacheFactory = $memCacheFactory;
-		$this->dispatcher = $dispatcher;
-		$this->logger = $logger;
+	public function __construct(
+		private IUserSession $userSession,
+		private IConfig $config,
+		private AppConfig $appConfig,
+		private IGroupManager $groupManager,
+		private ICacheFactory $memCacheFactory,
+		private EventDispatcherInterface $legacyDispatcher,
```
[the remark this answers is not in the document]
Comment (shdehnavi) [reply lib/private/App/AppManager.php]: It's addressed in the next commit

**Comment by shdehnavi** [comment]
@blizzz All conflicts have been resolved, Ready to merge.

**Review by fsamapoor — APPROVED** [review]

**Pull request merged**